dask/array/core.py

```diff
@@ -1404,7 +1404,11 @@ def chunks(self, chunks):
     def __len__(self):
         if not self.chunks:
             raise TypeError("len() of unsized object")
-        return sum(self.chunks[0])
+        try:
+            return int(sum(self.chunks[0]))
+        except (TypeError, ValueError):
+            pass
+        raise TypeError("len() of object with unknown size is unsupported.")
```

Comment:
Nitpick: Perhaps it'd be more appropriate for this to be a `ValueError` instead of a `TypeError`?

Reply:
"object with unknown size" might be unclear to some users. What about "len() of a Dask Array with unknown chunk sizes is unsupported.", which is a bit more explicit, instead?

Reply:
We could also reuse this message 

https://github.com/dask/dask/blob/d3f7cd248fd5060ba882804f89848f669ece93e2/dask/array/core.py#L83-L90

to point users to the `compute_chunk_sizes()` method as a possible solution if they indeed do want to get the `len` of a Dask Array

Reply:
Thanks for pointing out `unknown_chunk_message`, I agree it'd be good to use here.